recipes/opene57/all/conanfile.py

```diff
@@ -0,0 +1,135 @@
+from conans import ConanFile, tools, CMake
+from conan.tools.microsoft import msvc_runtime_flag
+from conans.errors import ConanInvalidConfiguration
+import os
+
+class Opene57Conan(ConanFile):
+    name = "opene57"
+    description = "A C++ library for reading and writing E57 files, " \
+                  "fork of the original libE57 (http://libe57.org)"
+    topics = ("e57", "libe57", "3d", "astm")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/openE57/openE57"
+    license = ("MIT", "BSL-1.0")
+    settings = "os", "compiler", "arch", "build_type"
+    options = { "with_tools": [True, False],
+                "shared": [True, False],
+                "fPIC": [True, False]
+               }
+    default_options = {
+                "with_tools": False,
+                "shared": False,
+                "fPIC": True
+               }
+    generators = "cmake", "cmake_find_package"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "7",
+            "clang": "6",
+            "apple-clang": "10",
+        }
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        self.options['xerces-c'].shared = self.options.shared
+
+        if self.settings.os == "Linux" or tools.is_apple_os(self.settings.os):
+            self.options['icu'].shared = self.options.shared
+
+        if self.options.with_tools:
+            self.options['boost'].shared = self.options.shared
+            self.options['boost'].multithreading = True
```

Comment:
why shared dependencies if shared and static dependencies if static?

Reply:
At the moment, the library can only be built as static. And if "shared" is True, the user will be warned that the configuration isn't supported at the moment

Reply:
It doesn't answer to my question actually. Why do you force shared option of dependencies?

Reply:
i don't have an answer for it. The original library was using all static libraries and linking as static itself, so I try to enforce this condition here.

Reply:
Please remove this logic, it's not required.